src/rgw/rgw_lua_background.cc

```diff
@@ -153,7 +154,44 @@ void Background::run() {
       }
       ldpp_dout(dpp, 10) << "Lua background thread resumed" << dendl;
     }
-    
+    std::size_t max_memory = cct->_conf->rgw_lua_max_memory_per_state;
+    std::uint64_t max_runtime = cct->_conf->rgw_lua_max_runtime_per_state;
+    if (lguard->get_max_memory() != max_memory) {
+      auto res = lguard->set_max_memory(max_memory);
+      if (res == -1) {
+        ldpp_dout(dpp, 10)
+            << "Lua state required reset due to memory limit change." << dendl;
+        lguard.reset();
+        lguard = std::make_unique<lua_state_guard>(
+            cct->_conf->rgw_lua_max_memory_per_state,
+            cct->_conf->rgw_lua_max_runtime_per_state, dpp);
+        L = lguard->get();
+        if (!L) {
+          ldpp_dout(dpp, 1)
+              << "Failed to create state for Lua background thread" << dendl;
+          return;
+        }
+        try {
+          open_standard_libs(L);
+          set_package_path(L, lua_manager->luarocks_path());
+          create_debug_action(L, cct);
+          create_background_metatable(L);
+        } catch (const std::runtime_error& e) {
+          ldpp_dout(dpp, 1) << "Failed to create initial setup of Lua "
+                               "background thread. error "
+                            << e.what() << dendl;
+          return;
+        }
```

Comment:
nit: can you split this section into its own function and reuse it in line 127 - 144 ?